File: src/main/java/org/taktik/icure/dao/AgendaDAO.java
package org.taktik.icure.dao;

import org.taktik.icure.entities.Agenda;

import java.util.List;

public interface AgendaDAO extends GenericDAO<Agenda> {
    List<Agenda> getAllAgendaForUser(String userId);

    List<Agenda> getReadableAgendaForUser(String userId);
}

File: src/main/kotlin/org/taktik/icure/asyncdao/AgendaDAO.kt
package org.taktik.icure.asyncdao

import kotlinx.coroutines.flow.Flow
import org.taktik.icure.entities.Agenda
import java.net.URI

interface AgendaDAO: GenericDAO<Agenda> {
    fun getAllAgendaForUser(userId: String): Flow<Agenda>

    fun getReadableAgendaForUser(userId: String): Flow<Agenda>
}
